File: pinyougou/pinyougou-manage-web/src/main/webapp/js/controller/brandController.js
//定义处理器
app.controller("brandController", function ($scope,$controller,brandService) {

    //继承处理器
    $controller("baseController", {$scope:$scope});

    $scope.findAll = function () {
        //发送请求到后台获取数据
        brandService.findAll().success(function (response) {
            //将返回结果设置到一个上下文变量中
            $scope.list = response;
        }).error(function () {
            alert("加载数据失败");
        });

    };

    //根据分页信息查询数据
    $scope.findPage = function (pageNo, rows) {
        brandService.findPage(pageNo,rows).success(function (response) {
            //response是分页对象（列表，总记录数）
            $scope.list = response.rows;
            //给分页组件重新设置最新的总记录数
            $scope.paginationConf.totalItems = response.total;
        });

    };


    //修改后保存
    $scope.save = function () {
        var obj;
        if($scope.entity.id != null) {
            //修改
            obj = brandService.update($scope.entity);
        }else {
            //添加
            obj = brandService.add($scope.entity);
        }

        obj.success(function (response) {
            if(response.success) {
                //如果操作成功,则刷新列表
                $scope.reloadList();
            }else {
                alert(response.message);
            }
        });
    };

    //根据主键查询
    $scope.findOne = function (id) {
        brandService.findOne(id).success(function (response) {
            $scope.entity = response;
        });
    };

    // 批量删除
    $scope.delete = function () {
        if ($scope.selectedIds.length < 1) {
            alert(" 请选择要删除的记录");
            return;
        }
        if(confirm(" 确定要删除选中的记录吗？")){
            brandService.delete($scope.selectedIds).success(function (response) {
                if(response.success){
                    $scope.reloadList();
                    $scope.selectedIds = [];
                } else {
                    alert(response.message);
                }
            });
        }
    };

    // 搜索
    // 定一个空的搜索条件对象
    $scope.searchEntity = {};
    $scope.search = function (page, rows) {
        brandService.search(page,rows,$scope.searchEntity).success(function (response) {
            $scope.list = response.rows;
            $scope.paginationConf.totalItems = response.total;
        });
    };
});

File: pinyougou/pinyougou-manage-web/src/main/java/com/pinyougou/manage/controller/BrandController.java
package com.pinyougou.manage.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;
import com.pinyougou.vo.PageResult;
import com.pinyougou.vo.Result;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@RequestMapping("/brand")
//@Controller
@RestController  //组合注解 Controller 和 ResponseBody；类上面的该注解对所有方法生效
public class BrandController {

    //从注册中心引入服务代理对象
    @Reference
    private BrandService brandService;


    /**
     * 接收品牌数据并保存品牌
     * @param brand 品牌数据
     * @return 操作结果
     */
    @PostMapping("/add")
    public Result add(@RequestBody TbBrand brand) {
        try {
            brandService.add(brand);
            return Result.ok("新增品牌成功");
        } catch (Exception e) {
            e.printStackTrace();
        }
        return Result.fail("新增品牌失败");
    }

    /**
     * 根据主键查询品牌
     * @param id
     * @return
     */
    @GetMapping("/findOne")
    public TbBrand findOne(long id) {
        return brandService.findOne(id);
    }
    /**
     * 修改后并保存
     * @param brand
     * @return
     */
    @PostMapping("/update")
    public Result update(@RequestBody TbBrand brand) {
        try {
            brandService.update(brand);
            return Result.ok("修改品牌成功");
        } catch (Exception e) {
            e.printStackTrace();
        }
        return Result.fail("修改品牌失败");
    }

    @GetMapping("/delete")
    public Result delete(Long[] ids) {
        try {
            brandService.deleteByIds(ids);
            return Result.ok(" 删除成功");
        } catch (Exception e) {
            e.printStackTrace();
        }
        return Result.fail(" 删除失败");
    }

    /**
     * 根据分页条件查询，查询第1页每页5条品牌列表
     * @param page 页号
     * @param rows 页大小
     * @return 品牌列表
     */
    @GetMapping("/testPage")
    public List<TbBrand> testPage(@RequestParam(value = "page",defaultValue = "1")Integer page,
                                  @RequestParam(value = "rows",defaultValue = "10")Integer rows) {
        //return brandService.testPage(page, rows);
        return (List<TbBrand>) brandService.findPage(page,rows).getRows();
    }

    //@RequestMapping(value = "/findAll",method = RequestMethod.GET)
    @GetMapping("/findAll")
    //@ResponseBody
    public List<TbBrand> findAll() {
        //return brandService.queryAll();
        return brandService.findAll();
    }

    //分页全部数据
    @GetMapping("/findPage")
    public PageResult findPage(@RequestParam(value = "pageNo",defaultValue = "1")Integer pageNo,
                               @RequestParam(value = "rows",defaultValue = "10")Integer rows) {
        return brandService.findPage(pageNo,rows);
    }

    /**
     * 根据分页条件查询
     * @param pageNo 页号
     * @param rows 页大小
     * @return 分页对象
     */
    @PostMapping("/search")
    public PageResult search(@RequestBody TbBrand brand,
                             @RequestParam(value="pageNo", defaultValue = "1")Integer pageNo,
                             @RequestParam(value="rows", defaultValue = "10")Integer rows){
        return brandService.search(brand, pageNo, rows);
    }

    /**
     * 用于模板管理的关联品牌下拉条功能,返回的数据格式符合 select2 格式
     * @return
     */
    @GetMapping("/selectOptionList")
    public List<Map<String, String>> selectOptionList() {
        return brandService.selectOptionList();
    }
}
